### PR TITLE
fix(settings): changing language sets title to "Add Friend"

### LINKED ISSUE #3708: Changing language set title to "Add Friend"
##### Brief Description

OS: Arch Linux
qTox version: v1.4.0-2-g24a7f4d.
Commit hash: 24a7f4d51960a73cb7234cec9a4a2811e9175c6c
toxcore: 0.0.0
Qt: 5.6.0

Reproducible: Always
##### Steps to reproduce
1. Setup one window mode
2. Change langugage
##### Observed Behavior

Window title changed on `qTox - Add friend`
##### Expected Behavior

Title should be `qTox - Settings`

### PR DESCRIPTION
**Pull request by ezavod**
This fixes #3708.
Changing language sets title of settings widget to "Settings" instead
of wrong "Add Friend".

Note that there is an other user who committed similar changes to his fork, but he has not been active for over half a year now, so I think its OK to get this minor bug fixed.

---
This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/4340)

### CODE REVIEW

**Comment by Diadlo** [comment]
Sorry, but this solution will produce #3100
So it's not fix the problem

**Comment by ezavod** [comment]
@Diadlo Sorry, did not noticed that.

Issue #3100 should not happen anymore.
So on startup title is translated as "Add Friend". Only after settings are visited at least once, title is translated as "Settings".

**Comment by Diadlo** [comment]
I don't think, that it's right solution. Can you replace it on current widget detection + set title in accordance with it?

**Comment by ezavod** [comment]
Did some code cleanup and added explicit check if settings are currently shown.
I don not think that it is necessary to check for other possible widgets, because title translation is handled correctly under all other circumstances except one changes the language from within a widget. That is only possible from settings.

**Comment by Diadlo** [comment]
@ezavod What if we will add new place to change language? For example it will change according to system language change? Sould we add more if's?

**Comment by ezavod** [comment]
@Diadlo 

> What if we will add new place to change language?

The way how it is currently handled by qTox felt a bit weird to me, but I have no idea how this could be improved. This is probably better for someone, who has a deeper understanding of how the qTox code works as a whole, than I have at the moment.

> change according to system language change?

I think this would also be a big task to do, because there are some problems.
For example: The user has the option to change the language in settings. So if a user sets the language to `A` in settings but the system-language is set to `B` should one use the user setting from qTox or the system setting?
A solution could be to add a new option to the language menu that is something like `Use system language` (and maybe make it default on fresh installs [see #2758]).

> Sould we add more if's?

No, this is not required at the moment. All titles are translated properly if the currently visible form/widget is changed.
This breaks if the language is changed from within a form/widget, because the title was set when the form/widget was set as active, but since this point in time the language was changed, but the title was not.
The only way this could happen is if the user changes the language from settings. Therefor it is sufficient to check if settings are visible to filter the language set event emitted at startup from user changes.
There is no option to change the language in one of the other possible forms/widgets (`AddFriendForm`, `GroupInviteForm`, `ProfileForm`, `FilesForm`,  `GenericChatroomWidget`) so there is no need to test for them.

**In short:** I think one should use this as a fix for #3708 (of course only if this works for everyone).
Refactoring the whole widget translation or detecting system-language is out of the scope of this PR, I think.

**Comment by sudden6** [comment]
@Diadlo @ezavod Should this be merged for 1.10?

**Comment by Diadlo** [comment]
I don't think, that it's the right way. But can be merged as hotfix

**Comment by sudden6** [comment]
ok, I'll merge it then.

**Comment by sudden6** [comment]
:lgtm_strong:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/4340#-:-KjSNJ0jATFWXGwbjtIR:b-jbsf8z)*

**Comment by ezavod** [comment]
> I don't think, that it's the right way. But can be merged as hotfix

This was meant as hotfix, so that the uses is not bothered with strange title changes anymore. Sorry that I can not do more, due to a lag of time and qTox code knowledge.
